src/Lightweight/SqlStatement.hpp

```diff
@@ -321,6 +321,67 @@ class [[nodiscard]] SqlResultCursor
     SqlStatement* m_stmt;
 };
 
+// input iterator
+template <typename T>
+class iterator
+{
+  public:
+    using difference_type = bool;
+    using value_type = T;
+
+    iterator(SqlStatement& stmt, std::string_view table_name):
+        m_stmt(&stmt),
+        m_table_name(table_name)
+    {
+    }
+
+    iterator operator++()
+    {
+        if (!m_stmt->FetchRow())
+            m_is_end = true;
```

Comment:
```suggestion
        m_is_end = !m_stmt->FetchRow();
```
Either this way, or early abort if we already reached the end.

Another topic, on the `m_` prefix. I actually thought about slightly moving towards how we name things in Contour, that is, using `_` preix for private member naming and continue in camelCase then (which also "slightly" matches what we use in the company). What do you think?

Reply:
I agree, initially i used `_` as a prefix and then changed it to `m_` xD, will change it back